### ISSUE TITLE
Docker build fails if gitea is a git submodule

### ISSUE THREAD

**Issue by AbdulrhmnGhanem** [go-gitea/gitea#20055]
### Description

To reproduce

```console
mkdir test
cd test
git init .
git submodule add https://github.com/go-gitea/gitea
cd gitea
docker build . -t tag
```
output :point_down:
```
error obtaining VCS status: exit status 128
	Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
	Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
	Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
	Use -buildvcs=false to disable VCS stamping.
Running go generate...
go: downloading github.com/mattn/go-sqlite3 v1.14.12
make: *** No rule to make target 'modules/public/bindata.go', needed by 'gitea'.  Stop.
The command '/bin/sh -c if [ -n "${GITEA_VERSION}" ]; then git checkout "${GITEA_VERSION}"; fi  && make clean-all build' returned a non-zero code: 2
```

### Gitea Version

cb50375

### Can you reproduce the bug on the Gitea demo site?

No

### Git Version

2.36.1

### Operating System

Linux

### How are you running Gitea?

Docker build.

**Comment by kasbah** [comment]
This golang issue may be relavant: https://github.com/golang/go/issues/51748

**Comment by lunny** [comment]
Closed because https://github.com/golang/go/issues/51748 resolved in go 1.18
